### PR TITLE
Add a gvcf file generated by sample, and others minors changes (merge on exome.bed, option to choose the type exon/transcript/gene of studied feature in GTF). 

### PR DESCRIPTION
**Pull request by SarahMaman**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnavar/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/rnavar _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Hi @SarahMaman I change the base from `master` to `dev`.
Thanks for adding this.
I was hoping to find some time to work on this, but you were more efficient

**Comment by SarahMaman** [comment]
Change the target branch to dev

**Comment by maxulysse** [comment]
@nf-core-bot fix linting pretty please :pray:

**Review by maxulysse — APPROVED** [review]

File: conf/modules.config [hunk 1]
```diff
@@ -316,6 +317,22 @@ process {
...
+        ].join(' ').trim()
+        ext.prefix = {"${meta.id}.haplotypecaller"}
+        publishDir  = [ enabled: false ]
+    }
+    
+    withName: GATK4_INDEXFEATUREFILE {
+        publishDir  = [
+            enabled: false
+        ]
+    }
```
Comment (maxulysse) [inline conf/modules.config:335]:
```suggestion
    
    withName: GATK4_HAPLOTYPECALLERGVCF {
        ext.args    = [
            '--dont-use-soft-clipped-bases',
            params.gatk_hc_call_conf ? "--standard-min-confidence-threshold-for-calling $params.gatk_hc_call_conf" : '',
            params.bam_csi_index ? "--create-output-variant-index false" : "", "-ERC GVCF"
        ].join(' ').trim()
        ext.prefix = {"${meta.id}.haplotypecaller"}
        publishDir  = [ enabled: false ]
    }
    
    withName: GATK4_INDEXFEATUREFILE {
        publishDir  = [
            enabled: false
        ]
    }
```

File: conf/modules.config [hunk 2]
```diff
@@ -316,6 +317,22 @@ process {
...
+    withName: GATK4_HAPLOTYPECALLERGVCF {
+        ext.args    = [
+            '--dont-use-soft-clipped-bases',
+            params.gatk_hc_call_conf ? "--standard-min-confidence-threshold-for-calling $params.gatk_hc_call_conf" : '',
+            params.bam_csi_index ? "--create-output-variant-index false" : "", "-ERC GVCF"
+        ].join(' ').trim()
+        ext.prefix = {"${meta.id}.haplotypecaller"}
+        publishDir  = [ enabled: false ]
+    }
+
```
Comment (maxulysse) [inline conf/modules.config:330]:
```suggestion
```

File: modules/local/gtf2bed.nf [hunk 3]
```diff
@@ -5,11 +5,12 @@ process GTF2BED {
...
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/r-base:3.5.0' :
-        'quay.io/biocontainers/r-base:3.5.0' }"
+        'quay.io/biocontainers/r-base:3.5.0'}"
 
     input:
     path gtf
-
+    val feature_type
+
```
Comment (maxulysse) [inline modules/local/gtf2bed.nf:13]:
```suggestion

```

File: modules/local/gtf2bed.nf [hunk 4]
```diff
@@ -18,15 +19,20 @@ process GTF2BED {
...
+    def allowed_type = ["exon", "transcript", "gene"];
+    if (feature_type){
+        feature_type = allowed_type.contains(feature_type) ? feature_type : "exon"
+    }
+    
     """
     Rscript --no-save -<<'RCODE'
         gtf = read.table("${gtf}", sep="\t")
-        gtf = subset(gtf, V3 == "exon")
+        gtf = subset(gtf, V3 == "${feature_type}")
```
Comment (maxulysse) [inline modules/local/gtf2bed.nf:30]:
```suggestion
        gtf = subset(gtf, V3 == "${feature_type}")
```

**Comment by maxulysse** [comment]
we'll need to fix nf-core linting, but it can be done in a separate PR, we're due for some TEMPLATE sync anyway

**Pull request merged**